Sources/Core/IStream.cpp

```diff
@@ -26,7 +26,7 @@
 #include "IStream.h"
 
 namespace spades {
-	IStream::~IStream() {}
+	IStream::~IStream() throw(Exception) {}
```

Comment:
I don't think using exceptions specifications is a good idea. They were deprecated in C++11 and removed in C++17: https://en.cppreference.com/w/cpp/language/except_spec

Reply:
I guess I missed that part. As a related matter, there’s #780 regarding deprecating exceptions thrown by assertion failures.